tests/test_mesh.py

```diff
@@ -130,7 +130,8 @@ def test_structured_mesh_from_coords():
         structured_coords=[range(1, 5), range(1, 4), range(1, 3)], structured=True
     )
     assert sm.dims == [0, 0, 0, 3, 2, 1]
-    assert_array_equal(sm.structured_coords, [range(1, 5), range(1, 4), range(1, 3)])
+    for obs, exp in zip(sm.structured_coords, [range(1, 5), range(1, 4), range(1, 3)]):
+        assert_array_equal(obs, exp)
```

Comment:
I guess if you value detailed error reporting, your approach is good to go. However, for simplicity, I am suggesting this, though it's not mandatory.
```suggestion
    assert sm.structured_coords == [range(1, 5), range(1, 4), range(1, 3)]
```